pkg/config/local.go

```diff
@@ -0,0 +1,90 @@
+package config
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/pkg/utils"
+	yaml "gopkg.in/yaml.v3"
+)
+
+var (
+	PlatformConfigFile = path.Join(GetDataHome(), "platform.yaml")
+)
+
+type PlatformInfo struct {
+	Current  types.Platform `yaml:"current"`
+	Previous types.Platform `yaml:"previous"`
+}
+
+func (p *PlatformInfo) Update(platform types.Platform) error {
+	if err := p.Load(); err != nil {
+		return err
+	}
+	if p.Current == "" {
+		p.Current = platform
+	}
+	p.Previous = p.Current
+	p.Current = platform
+
+	f, err := os.Create(PlatformConfigFile)
+	if err != nil {
+		return fmt.Errorf("error creating file %s: %v", PlatformConfigFile, err)
+	}
+	defer f.Close()
+	e := yaml.NewEncoder(f)
+	if err = e.Encode(p); err != nil {
+		return fmt.Errorf("error saving file: %s: %v", PlatformConfigFile, err)
+	}
+	return nil
+}
+
+func (p *PlatformInfo) Load() error {
+	data, err := ioutil.ReadFile(PlatformConfigFile)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("error loading %s: %v", PlatformConfigFile, err)
+	}
+	if data != nil {
+		decoder := yaml.NewDecoder(bytes.NewReader(data))
+		if err = decoder.Decode(p); err != nil {
+			return fmt.Errorf("error decoding %s: %v", PlatformConfigFile, err)
+		}
+	}
+	return nil
+}
+
+var (
+	Platform string
```

Comment:
What's the significance of this variable? (Otherwise all looks good!)

Reply:
It is bound by a CLI option introduced here named `--platform` (similar to `--kubeconfig` or `--context`).